### PR TITLE
Update emperor

### LINKED ISSUE #6: SVG Export
Export plot as svg image

### LINKED ISSUE #35: Error in coloring due to non alphanumeric characters
When the categories of a column in the mapping file has no alphanumeric characters (for example '), the coloring by metadata doesn't work.

### LINKED ISSUE #40: E-vident should use the Emperor framework
The python code that writes the HTML document in Evident is now being implemented in a more cohesive manner in qiime/emperor. Once Emperor, reaches a more stable state on it's own, E-vident should take advantage of it.

This should probably be a git [submodule](https://www.kernel.org/pub/software/scm/git/docs/git-submodule.html) inside E-vident.

### PR DESCRIPTION
**Pull request by antgonza**
This closes #40, resolves #35, solves #12, and closes #6

### CODE REVIEW

**Comment by ElDeveloper** [comment]
I'll be taking a look at this PR, however I think I won't be able to do it for the next few hours as I'm re-downloading the repository and I am on a very slow internet connection.

Just wanted to give a heads up to all the other devs.

**Comment by ElDeveloper** [comment]
This is looking great so far, amazed by the fact that it wasn't that difficult to integrate Emperor. So far just a couple things that I've noticed:
- Trying to visualize Fierer et al with 228 sequences per sample, 3/3 subjects, 9/31 samples per subject  and 3/10 iterations throws the following error:

``` python
MOD_PYTHON ERROR

ProcessId:      3646
Interpreter:    'evident'

ServerName:     'evident'
DocumentRoot:   '/var/www'

URI:            '/~evident/qiime_software/evident/evident/www/handler.psp'
Location:       None
Directory:      None
Filename:       '/home/evident/shared/qiime_software/evident/evident/www/handler.psp'
PathInfo:       ''

Phase:          'PythonHandler'
Handler:        'mod_python.psp'

Traceback (most recent call last):

  File "/usr/lib/python2.7/dist-packages/mod_python/importer.py", line 1537, in HandlerDispatch
    default=default_handler, arg=req, silent=hlist.silent)

  File "/usr/lib/python2.7/dist-packages/mod_python/importer.py", line 1229, in _process_target
    result = _execute_target(config, req, object, arg)

  File "/usr/lib/python2.7/dist-packages/mod_python/importer.py", line 1128, in _execute_target
    result = object(arg)

  File "/usr/lib/python2.7/dist-packages/mod_python/psp.py", line 337, in handler
    p.run()

  File "/usr/lib/python2.7/dist-packages/mod_python/psp.py", line 243, in run
    exec code in global_scope

  File "/home/evident/shared/qiime_software/evident/evident/www/handler.psp", line 101, in <module>
    tree_object=tree_object)

TypeError: generate_pcoa_cloud_from_point_in_omega() got an unexpected keyword argument 'map_headers'
```
- The Demo of the Costello, Turbaugh  datasets throws the following error:

``` python

MOD_PYTHON ERROR

ProcessId:      3646
Interpreter:    'evident'

ServerName:     'evident'
DocumentRoot:   '/var/www'

URI:            '/~evident/qiime_software/evident/evident/www/handler.psp'
Location:       None
Directory:      None
Filename:       '/home/evident/shared/qiime_software/evident/evident/www/handler.psp'
PathInfo:       ''

Phase:          'PythonHandler'
Handler:        'mod_python.psp'

Traceback (most recent call last):

  File "/usr/lib/python2.7/dist-packages/mod_python/importer.py", line 1537, in HandlerDispatch
    default=default_handler, arg=req, silent=hlist.silent)

  File "/usr/lib/python2.7/dist-packages/mod_python/importer.py", line 1229, in _process_target
    result = _execute_target(config, req, object, arg)

  File "/usr/lib/python2.7/dist-packages/mod_python/importer.py", line 1128, in _execute_target
    result = object(arg)

  File "/usr/lib/python2.7/dist-packages/mod_python/psp.py", line 337, in handler
    p.run()

  File "/usr/lib/python2.7/dist-packages/mod_python/psp.py", line 243, in run
    exec code in global_scope

  File "/home/evident/shared/qiime_software/evident/evident/www/handler.psp", line 131, in <module>
    map_headers, map_data)

TypeError: make_pcoa_plot() takes exactly 3 arguments (6 given)

```
- The HMPV13 demo colors first by description, might be worth to remove that column to improve the loading time, we should be able to use the body site.

Also, there are far too many files in this changeset to efficiently look at them all. Can you point at the ones that I should keep an eye open? I've noticed `UPDATE_EMPEROR.md` and `evident/pcoa.py` any others I should check?

Thanks!

**Comment by ElDeveloper** [comment]
Ok, just followed @antgonza's offline suggestions of: restarting apache and cleaning up all .pyc files and it seems to be working fine now.

However I'm now seeing this problem where the `PCoA` and `Parallel` buttons are hidden.

captura de pantalla 2013-08-08 a la s 20 22 51

**Comment by ElDeveloper** [comment]
I think it would probably make sense to use a `git-submodule` for Emperor. What does the other devs think? 

A quote from [kernel.org](https://www.kernel.org/pub/software/scm/git/docs/git-submodule.html):

> Submodules allow foreign repositories to be embedded within a dedicated subdirectory of the source tree, always pointed at a particular commit.

**Comment by wdwvt1** [comment]
@eldeveloper and all-knowing wikipedia explained this concept to me and it
seems like a great way to go.

**Comment by antgonza** [comment]
In my opinion it doesn't: the repository is fully copied, you can not just
submodule a folder, thus we will need to change all the code to point to
point to the new folders and host, twice, all the files.

**Comment by ElDeveloper** [comment]
Hello,

Ah, I see what you are saying. I most certainly think you are right but cluttering the repository with changes from another repository sounds like a bad idea and kind of hard to maintain. 

I think it would probably be worth to try and make it work like that for future implementations and maybe deal with this as it is right now?

An alternative is if Evident writes the header instead of the constant EMPEOR_HTML_HEADER, then we wouldn't have to change the paths and it would basically just point to that submodule.

Thoughts?

**Comment by antgonza** [comment]
Not sure why you think this is messy? We are only copying one folder.
The rest of the files come from installing Emperor. In fact, we could
even set up a global variable to let the system know where Evident is
and avoid having to copy that folder but this can be problematic as
for the way is set up the web libraries should be static. Is the idea
to avoid having to install Emperor and just have it as its? Not sure
if this will work because you still will need to set up the paths ...

BTW the main files changed are:
UPDATE_EMPEROR.md
evident/pcoa.py
www/handler.psp
www/index.html

Deleted as we are now importing these functionalities from Emperor:
tests/test_evident.py
tests/test_pcoa.py

**Comment by ElDeveloper** [comment]
I think it's messy because everytime you update the resources, for emperor, you will clutter the repo with things that are not relevant to itself. Whilst if you have a submodule, you should be able to easily update at what version you want to point.

I agree with you that the global variable would be problematic.

No, my idea is not to avoid installing emperor, emperor will have to be installed in the system. I just think that if we had this a submodule, it would become easier and cleaner to change versions of emperor.

Thanks for pointing those files out, taking a look at them right away.

File: evident/pcoa.py [hunk 1]
```diff
@@ -14,101 +14,51 @@
 """code for doing pcoa analysis"""
 
 from numpy import array
-
-from evident import make_pcoa_plot
+from cStringIO import StringIO
```
Comment (ElDeveloper) [inline evident/pcoa.py:17]: Any reason to use `cStringIO` instead of `StringIO` or is this a typo?

**Comment by ElDeveloper** [comment]
I think everything is looking good besides that minor comment, which is a question really.

File: evident/pcoa.py (hunk 1 above)
Comment (antgonza) [inline evident/pcoa.py:17]: No reason really but it sounds like in general we should use cStringIO
based on speed:
http://docstore.mik.ua/orelly/other/python/0596001886_pythonian-chp-10-sect-5.html

**Comment by antgonza** [comment]
So what will be the plan with this? Add an issue to make it work with
submodules? BTW I looked into this option for this pull request but it
didn't make sense to me based on the duplication of code/scripts, and
having those copied files exposed in the web as they have to be static
(which will mean that the full submodule will have to be exposed).
However, I agree that will make it cleaner. What to do in this
conundrum?

File: evident/pcoa.py (hunk 1 above)
Comment (ElDeveloper) [inline evident/pcoa.py:17]: Excellent, good to know!

**Comment by ElDeveloper** [comment]
I've opened a new issue for the usage of submodules #45.

**Comment by ElDeveloper** [comment]
Should we prepare a new virtual machine with this build?

**Comment by antgonza** [comment]
That's a good idea but ideally we should create a list of commands or a GUI
to auto update the machine.

**Comment by ElDeveloper** [comment]
Agreed, perhaps having a script that everytime the virtual machine boots, it tries to pull the latest version of the repository? I think VirtualBox offers a way to automate the creation of a virtual machine, but I don't know how we would go as to configure the machine itself.

**Comment by antgonza** [comment]
At boot is an option but most of the time the Virtual Machine has no
internet the first time it boots so maybe having a command or a GUI is the
way to go.

**Pull request merged**